### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by kiboma2021**
- Add Redux Toolkit(npm install react-redux @reduxjs/toolkit).
- Structure your application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files. 
- Write your book's actions and reducer (in /src/redux/books/books.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
- Define action types for adding and removing a book.
- Set the initial state to be an empty array of books.
- Export Action Creators for your actions.
- Write your reducer and export it as default.
- Define state changes for the actions that you created.
- In case of unknown action - return the current state.
- Write your categories actions and reducer (in /src/redux/categories/categories.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
- Define an action type for checking the status.
- Set the initial state to be an empty array of categories.
- Export Action Creators for your actions.
- Write your reducer and export it as default.
- For the check status action return a string "Under construction".
- In case of unknown action - return the current state.
- Configure the Redux Store (/src/redux/configureStore.js):
- Importing the necessary methods from Redux Toolkit.
- Combine both reducers into a root reducer by using configureStore function.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "react-dom": "^18.2.0",
     "react-router-dom": "^6.3.0",
     "react-scripts": "5.0.1",
+    "redux": "^4.2.0",
     "web-vitals": "^2.1.4"
```
Comment (DammyShittu) [inline package.json]: - [ ] Ensure you install the `redux toolkit` package. It should be present in this your package.json file.

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const ADD_BOOK = 'bookstore/books/ADD_BOOK';
+const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
+
+export const addBook = (book) => ({
+  type: ADD_BOOK,
+  id: book.id,
```
Comment (DammyShittu) [inline src/redux/books/books.js]: - [ ] I don't think we are saving only the book id. We need to store the entire book as well - so there's no need to extract only the id here. We can just have something like this for example (you are free to call `payload` anything you wish or even leave it as `book`):

  ```
  type: ADD_BOOK,
  payload: book
  ```

  Then in the reducer, you need to change `[...state, action.id]` to `[...state, action.payload]`

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+export const removeBook = (book) => ({
+  type: REMOVE_BOOK,
+  id: book.id,
+});
+
+const booksReducer = (state = [], action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [...state, action.id];
```
Comment (DammyShittu) [inline src/redux/books/books.js]: - [ ] Ensure you update based on any changes implemented above.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @kiboma2021,

✔️ File structure matches the ducks pattern

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json (hunk 1 above)
Comment (kiboma2021) [reply package.json]: @DammyShittu , this is done

File: src/redux/books/books.js (hunk 2 above)
Comment (kiboma2021) [reply src/redux/books/books.js]: The change has been implemented

File: src/redux/books/books.js (hunk 3 above)
Comment (kiboma2021) [reply src/redux/books/books.js]: This has been updated. Thanks

**Review by keroloussamy — APPROVED** [review]
Hi @kiboma2021, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️
- Add all the requirements. ✔️

clap

## Optional suggestions
_Check the comments under the review._

- [ ] x

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**